aunt-leahs-app/src/pages/AdminVolunteerDataPage.jsx

```diff
@@ -1,102 +1,147 @@
 import React, { useState, useEffect } from 'react';
-import { AzureAD } from 'react-aad-msal';
+import axios from 'axios';
+import moment from 'moment';
+import { AzureAD, withAuthentication } from 'react-aad-msal';
 
-import { withRouter } from 'react-router-dom';
-import { connect } from 'react-redux';
-import { compose } from 'redux';
 import AdminHeader from '../components/AdminHeader';
 import CustomTable from '../components/CustomTable';
 import CustomButton from '../components/CustomButton';
-import { setCurrentPage } from '../redux/page/pageAction';
-
-import { authProvider } from '../auth/authProvider';
+import { ExportToCsv } from 'export-to-csv';
+import { authProvider, authorizedFetch } from '../auth/authProvider';
 import store from '../redux/store';
+import { volunteerAPIBaseURL } from '../constants';
 
-function AdminShiftDataPage({ setCurrentPage }) {
+function AdminVolunteerDataPage({ setCurrentPage }) {
+	const [volunteerData, setVolunteerData] = useState(['']);
 
-	const [dateLastModifiedClear, setDateLastModifiedClear] = useState('');
-	const [dateLastModifiedExport, setDateLastModifiedExport] = useState('');
-	const [volunteerData, setVolunteerData] = useState(['']); // useState(constants.volunteerData);
-	//const volunteerData = constants.volunteerData;
+	const [adminHistory, setAdminHistory] = useState({
+		lastClearedTime: null,
+		lastExportedTime: null,
+	});
+	
+	const volunteersEndpointPath = '/api/volunteers';
+	const volunteerExportHistoryEndpoint = volunteerAPIBaseURL + '/api/history';
 
-	const exportData = (data) => {
-		alert('Exporting data...');
-	};
+	useEffect(() => {
+		getVolunteers();
+		getAdminHistory();
+	}, []);
 
-	const clearData = (data) => {
-		alert('Clearing data...');
-		clearVolunteers();
+	const options = {
+		fieldSeparator: ',',
+		filename: 'Volunteer Data',
+		quoteStrings: '"',
+		decimalSeparator: '.',
+		showLabels: true,
+		showTitle: false,
+		title: 'Volunteer Data',
+		useTextFile: false,
+		useBom: true,
+		useKeysAsHeaders: true,
+		// headers: ['Column 1', 'Column 2', etc...] <-- Won't work with useKeysAsHeaders present!
 	};
+	const csvExporter = new ExportToCsv(options);
 
+	const exportData = async () => {
+		try {
+			csvExporter.generateCsv(volunteerData);
 
-	useEffect(() => {
-		getVolunteers();
-	}, []);
+			await axios.put(volunteerExportHistoryEndpoint, {
```

Comment:
The history endpoints needs authentication

Reply:
Added